### PR TITLE
WandB Pipeline

### PR DESCRIPTION
**Pull request by FrancisPicard2000**
Weights and Biases (Wandb) Pipeline

Added two script arguments to main.py:
--wandb_run_name: specific name for the run (required)
--wandb_saving_model_name: artifact name to save the checkpoints (optional)

Each run monitors the Training & Validation accuracies on both tasks through 4 different plots that are updated after each epoch

The Overview sections contains
- The key hyperparameters under Config
- The key results on the Train/Valid/Test under Summary
- (Optionally) The artifact which has the checkpoints for the trained model (checkpoints are saved every 4 epochs and at the end under Files)

The Logs section contains
- The content printed to stdout
- The local logfile wasn't saving anything, everything is now printed to stdout
- I also changed the Total Execution Time format to HH:MM:SS

Additional Functional Changes
- I moved the test set evaluation outside the training loop in train.py, to avoid unnecessarily monitoring the testing loss
- Simplified a lot the local model saving, assuming it would be saved on Wandb as well

TODO
- The code works assuming only_evaluation is set to 3. I didn't try with the other options. 

Observations:
- Calling the "evaluate" function from SoccerNet.Evaluation.MV_FoulRecognition for the first time on the Validation and Test sets (in train.py) generated a RuntimeWarning: invalid value encountered in divide, I don't know if we can change this. Maybe this is why the test set evaluation was also under the training loop
- The "evaluate" function also prints <class 'numpy.ndarray'> to stdout, I don't know if we can change this. Maybe they tried to debug their code (first point above)

### CODE REVIEW

**Review by sergemalo — APPROVED** [review]
All good!
